projects/portal/src/app/pages/delegate/validators/validators.usecase.service.ts

```diff
@@ -19,70 +16,73 @@ import { concatMap, filter, map, mergeMap, withLatestFrom } from 'rxjs/operators
   providedIn: 'root',
 })
 export class ValidatorsUseCaseService {
-  private validatorsList$: Observable<InlineResponse20041Validators[] | undefined>;
-  private allValidatorsTokens$: Observable<number | undefined>;
-  private validatorsWithShare$: Observable<validatorWithShareType[]>;
-  private accAddress$: Observable<cosmosclient.AccAddress>;
-  private delegationsInService$: Observable<InlineResponse20038DelegationResponses[] | undefined>;
-  private delegatedValidatorsInService$: Observable<
-    (InlineResponse20041Validators | undefined)[] | undefined
-  >;
-  private totalDelegation$: Observable<number | undefined>;
-  private unbondingDelegations$: Observable<(InlineResponse20047 | undefined)[]>;
-
   constructor(
     private readonly walletService: WalletService,
     private readonly cosmosRest: CosmosRestService,
-  ) {
-    this.validatorsList$ = this.cosmosRest.getValidators$();
-    this.allValidatorsTokens$ = this.validatorsList$.pipe(
+  ) {}
+  get currentStoredWallet$(): Observable<StoredWallet | null | undefined> {
+    return this.walletService.currentStoredWallet$;
+  }
+  get validatorsList$(): Observable<InlineResponse20041Validators[] | undefined> {
+    return this.cosmosRest.getValidators$();
+  }
+  get accAddress$(): Observable<cosmosclient.AccAddress> {
+    return this.walletService.currentStoredWallet$.pipe(
+      filter((wallet): wallet is StoredWallet => wallet !== undefined && wallet !== null),
+      map((wallet) => cosmosclient.AccAddress.fromString(wallet.address)),
+    );
+  }
```

Comment:
Acquiring the current accAddress$ is implemented in various places, so I think it would be better if it could be transferred to walletService.

I'll leave it up to you to change it in this PR.

Reply:
Since there are many changes, I would like to avoid fixing them in this PR.